src/skarab_fpga.py

```diff
@@ -1531,6 +1600,62 @@ def _complete_sdram_configuration(self):
             raise SkarabSdramError(errmsg)
         LOGGER.info('Rebooting from SDRAM.')
 
+    def read_hmc_i2c(self, interface, slave_address, read_address, format_print=False):
+        """
+        Read a register on the HMC device via the I2C interface
+        Prints the data in binary (32-bit) and hexadecimal formats
+        Also returns the data
+        :param interface: identifier for i2c interface:
+                          0 - SKARAB Motherboard i2c
+                          1 - Mezzanine 0 i2c
+                          2 - Mezzanine 1 i2c
+                          3 - Mezzanine 2 i2c
+                          4 - Mezzanine 3 i2c
+        :param slave_address: I2C slave address of device to read
+        :param read_address: register address on device to read
+        :return: read data / None if fails
+        """
+
+        response_type = 'sReadHMCI2CResp'
+        expect_response = True
+
+        # handle read address (pack it as 4 16-bit words)
+        # TODO: handle this in the createPayload method
+        read_address = ''.join([struct.pack('!H', x) for x in
+                                struct.unpack('!4B', struct.pack('!I',
+                                                                 read_address))])
+
+        # create payload packet structure
+        request = sd.ReadHMCI2CReq(self.seq_num, interface, slave_address,
+                                   read_address)
+
+        # send payload and return response object
+        response = self.send_packet(payload=request.create_payload(),
+                                    response_type='ReadHMCI2CResp',
+                                    expect_response=True,
+                                    command_id=sd.READ_HMC_I2C,
+                                    number_of_words=15, pad_words=2)
+
+        if response is not None:
+            if response.read_success:
+                hmc_read_bytes = response.read_bytes  # this is the 4 bytes
+                # read
+                # from the register
+                # want to create a 32 bit value
+
+                hmc_read_word = struct.unpack('!I', struct.pack('!4B', *hmc_read_bytes))[0]
+                if format_print:
```

Comment:
@paulprozesky Yip. Initially, it was just print when the guys were debugging the HMC. An oversight made by myself.